### PR TITLE
chore: simplify construction event handlers + few functions

### PR DESCRIPTION
**Pull request by winex**
don't squash, please. best to be merged after #307 is in

_this is picked from destroy event fix #300, which depends on it_

### CODE REVIEW

File: construction-event-handler.lua
Comment (winex) [inline construction-event-handler.lua:1]: this diff looks horrible, it's better to view whole file

**Review by winex — COMMENTED** [review]
:+1:, though your `luacheck` hates me :p

**Comment by bahrmichael** [comment]
@winex I added `serpent` to the luacheckrc. You can pull the latest updates to make the build green :)

**Comment by winex** [comment]
@bahrmichael rebased, _coding speedrun is_ green now :)

**Review by bahrmichael — CHANGES_REQUESTED** [review]
Would it be possible to avoid goto statements? I'm not aware if there's anything requiring those, and prefer if/else statements over jumps across the code.

**Comment by winex** [comment]
> Would it be possible to avoid goto statements? I'm not aware if there's anything requiring those, and prefer if/else statements over jumps across the code.

yep, changed. `goto`'s aren't my preference as well, but sometimes it simplifies **cleanup** code paths to just few lines... that's why i used it here - it just states that you never forget one last thing(s).

though, as Lua doesn't have `continue` statement (lol) i still use close-jump `goto`s sometimes in loops to minimize `if-then` nesting

**Review by bahrmichael — APPROVED** [review]
Thank you for the refactoring!

**Pull request merged**